### PR TITLE
Add D22 models

### LINKED ISSUE #182: Add D22 and D22_MWAvg models
As soon as the NIR extinction paper gets accepted, the final NIR extinction curves should be added.

### PR DESCRIPTION
**Pull request by karllark**
Adds Decleir et al. (2022) MWAvg and R(V) dependent models.

Also makes a minor update to G21_MWAvg and minor text updates to references doc.

Closes #182

### CODE REVIEW

**Review by mdecleir — COMMENTED** [review]
A few small comments, otherwise looks good!

File: dust_extinction/averages.py [hunk 1]
```diff
@@ -1398,4 +1400,120 @@ def evaluate(self, in_x):
...
+        ValueError
+           Input x values outside of defined range
+        """
+        x = _get_x_in_wavenumbers(in_x)
+
+        # check that the wavenumbers are within the defined range
+        _test_valid_x_range(x, self.x_range, self.__class__.__name__)
+
+        # setup the model
+        d22_fit = PowerLaw1D(alpha=1.71, amplitude=0.386, x_0=1.0)
```
Comment (mdecleir) [inline dust_extinction/averages.py:1515]:
```suggestion
        d22_fit = PowerLaw1D(alpha=-1.71, amplitude=0.386, x_0=1.0)
```

File: dust_extinction/averages.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import numpy as np
 from scipy.interpolate import interp1d
 from astropy.table import Table
+from astropy.modeling.models import PowerLaw1D
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/averages.py:8]: Yes, but this is more generic.

File: dust_extinction/averages.py [hunk 3]
```diff
@@ -1398,4 +1400,120 @@ def evaluate(self, in_x):
         )
 
         # return A(x)/A(V)
+        # G21 a full dust_extinction model, hence send in x with units
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/averages.py:1403]: It is, mainly to keep me from making a mistake next time I look at it.

File: dust_extinction/averages.py [hunk 4]
```diff
@@ -1398,4 +1400,120 @@ def evaluate(self, in_x):
...
+        import astropy.units as u
+
+        from dust_extinction.averages import D22_MWAvg
+
+        mod = D22_MWAvg()
+
+        fig, ax = plt.subplots()
+
+        # generate the curves and plot them
+        lam = np.logspace(np.log10(0.8), np.log10(4.9), num=1000)
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/averages.py:1439]: Might work.  There is a check of the wavelength range and I've had some issues with being exactly at the min/max.  As this is just for a plot, decided to not worry about it.

File: docs/dust_extinction/choose_model.rst [hunk 5]
```diff
@@ -101,6 +104,8 @@ significantly more extinction curves than the
 +----------+-------------+-------------+------------------+--------------+
 | F19      |  R(V)       |   0.3 - 8.7 |      0.115 - 3.3 |           MW |
 +----------+-------------+-------------+------------------+--------------+
+| D22      |  R(V)       |  0.2 - 1.25 |        0.8 - 5.0 |           MW |
```
[the remark this answers is not in the document]
Comment (karllark) [reply docs/dust_extinction/choose_model.rst:107]: Fixed.

File: dust_extinction/averages.py [hunk 6]
```diff
@@ -1398,4 +1400,120 @@ def evaluate(self, in_x):
...
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+        import astropy.units as u
+
+        from dust_extinction.averages import D22_MWAvg
+
+        mod = D22_MWAvg()
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/averages.py:1434]: Nope.  Removed.

File: dust_extinction/averages.py [hunk 7]
```diff
@@ -1398,4 +1400,120 @@ def evaluate(self, in_x):
...
+    x_range = [1.0 / 5.0, 1.0 / 0.8]
+
+    Rv = 3.12
+
+    def __init__(self, **kwargs):
+
+        # get the tabulated information
+        data_path = pkg_resources.resource_filename("dust_extinction", "data/")
+
+        # GCC09 sigma clipped average of 13 diffuse sightlines
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/averages.py:1474]: Done.

File: dust_extinction/averages.py [hunk 8]
```diff
@@ -1398,4 +1400,120 @@ def evaluate(self, in_x):
...
+        x = _get_x_in_wavenumbers(in_x)
+
+        # check that the wavenumbers are within the defined range
+        _test_valid_x_range(x, self.x_range, self.__class__.__name__)
+
+        # setup the model
+        d22_fit = PowerLaw1D(alpha=1.71, amplitude=0.386, x_0=1.0)
+
+        # return A(x)/A(V)
+        # Note that model in D22 was done versus waelength in microns
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/averages.py:1518]: Fixed.

File: dust_extinction/parameter_averages.py [hunk 9]
```diff
@@ -1227,3 +1228,118 @@ def evaluate(self, in_x, Rv):
...
         return a_rV
+
+
+class D22(BaseExtRvModel):
+    r"""
+    Decleir et al (2022) extinction model calculation
+
+    Decleir, Gordon, et al. (2022, ApJ, submitted) model.
+    Based on a sample of stars observed spectroscopically in the
+    optical with IRTF/SpeX.
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/parameter_averages.py:1239]: Done.

File: dust_extinction/parameter_averages.py [hunk 10]
```diff
@@ -1227,3 +1228,118 @@ def evaluate(self, in_x, Rv):
...
+
+        # ensure Rv is a single element, not numpy array
+        Rv = Rv[0]
+
+        # intercepts
+        mod_a = PowerLaw1D(amplitude=0.377, alpha=1.78, x_0=1.0)
+        a = mod_a(1.0 / x)
+
+        # slopes
+        # from spline interpolation - need ot get this info
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/parameter_averages.py:1341]: Fixed.

File: dust_extinction/parameter_averages.py [hunk 11]
```diff
@@ -1227,3 +1228,118 @@ def evaluate(self, in_x, Rv):
...
+
+        ax.set_xlabel(r"$\lambda$ [$\mu m$]")
+        ax.set_ylabel(r"$A(x)/A(V)$")
+
+        ax.legend(loc="best")
+        plt.show()
+    """
+
+    Rv_range = [2.5, 5.5]
+    x_range = [1 / 4.0, 1 / 0.85]
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/parameter_averages.py:1287]: Yes it does.  Changed.

File: dust_extinction/parameter_averages.py [hunk 12]
```diff
@@ -1227,3 +1228,118 @@ def evaluate(self, in_x, Rv):
...
+
+        # intercepts
+        mod_a = PowerLaw1D(amplitude=0.377, alpha=1.78, x_0=1.0)
+        a = mod_a(1.0 / x)
+
+        # slopes
+        # from spline interpolation - need ot get this info
+        b = interpolate.splev(1.0 / x, self.spline_rep, der=0)
+
+        # return A(x)/A(55)
```
[the remark this answers is not in the document]
Comment (karllark) [reply dust_extinction/parameter_averages.py:1344]: Fixed.

**Pull request merged**